db/transaction_log_impl.cc

```diff
@@ -199,7 +199,7 @@ void TransactionLogIteratorImpl::NextImpl(bool internal) {
       if (current_last_seq_ == versions_->LastSequence()) {
         current_status_ = Status::OK();
       } else {
-        current_status_ = Status::Corruption("NO MORE DATA LEFT");
```

Comment:
Now that we correcting the Status type I wonder if we should correct the Status message as well. "NO MORE DATA LEFT" is quite misleading as there are actually more data to be read from the WAL. How about this message? "Create a new iterator to fetch the new tail."